packages/next/src/server/require-hook.ts

```diff
@@ -25,159 +23,33 @@ addHookAliases([
-          ),
-        ],
-        [
-          'react-server-dom-webpack/server.edge',
-          require.resolve(
-            `next/dist/compiled/react-server-dom-webpack-experimental/server.edge`
-          ),
-        ],
-        [
-          'react-server-dom-webpack/server.node',
-          require.resolve(
-            `next/dist/compiled/react-server-dom-webpack-experimental/server.node`
-          ),
-        ],
-      ])
-    } else {
-      addHookAliases([
-        ['react', require.resolve(`next/dist/compiled/react`)],
-        [
-          'react/package.json',
-          require.resolve(`next/dist/compiled/react/package.json`),
-        ],
-        [
-          'react/jsx-runtime',
-          require.resolve(`next/dist/compiled/react/jsx-runtime`),
-        ],
-        [
-          'react/jsx-dev-runtime',
-          require.resolve(`next/dist/compiled/react/jsx-dev-runtime`),
-        ],
-        [
-          'react-dom',
-          require.resolve(`next/dist/compiled/react-dom/server-rendering-stub`),
-        ],
-        [
-          'react-dom/package.json',
-          require.resolve(`next/dist/compiled/react-dom/package.json`),
-        ],
-        [
-          'react-dom/client',
-          require.resolve(`next/dist/compiled/react-dom/client`),
-        ],
-        [
-          'react-dom/server',
-          require.resolve(`next/dist/compiled/react-dom/server`),
-        ],
-        [
-          'react-dom/server.browser',
-          require.resolve(`next/dist/compiled/react-dom/server.browser`),
-        ],
-        [
-          'react-dom/server.edge',
-          require.resolve(`next/dist/compiled/react-dom/server.edge`),
-        ],
-        [
-          'react-server-dom-webpack/client',
-          require.resolve(`next/dist/compiled/react-server-dom-webpack/client`),
-        ],
-        [
-          'react-server-dom-webpack/client.edge',
-          require.resolve(
-            `next/dist/compiled/react-server-dom-webpack/client.edge`
-          ),
-        ],
-        [
-          'react-server-dom-webpack/server.edge',
-          require.resolve(
-            `next/dist/compiled/react-server-dom-webpack/server.edge`
-          ),
-        ],
-        [
-          'react-server-dom-webpack/server.node',
-          require.resolve(
-            `next/dist/compiled/react-server-dom-webpack/server.node`
-          ),
-        ],
-      ])
+    if (request === 'react-dom') {
+      packagePath = '/server-rendering-stub'
     }
-  } else {
-    addHookAliases([
-      ['react/jsx-runtime', require.resolve(`react/jsx-runtime`)],
-      ['react/jsx-dev-runtime', require.resolve(`react/jsx-dev-runtime`)],
-    ])
+
+    const aliasedPackage =
+      process.env.__NEXT_PRIVATE_PREBUNDLED_REACT === 'experimental'
+        ? `${packageName}-experimental`
+        : packageName
+    const mapped = require.resolve(
+      `next/dist/compiled/${aliasedPackage}${packagePath}`
```

Comment:
this will break the NFT file tracing as it won't include the react dist compiled anymore. I'm doing a workaround in https://github.com/vercel/next.js/pull/52206 

what's the reason for the simplication changes?